solidago/src/solidago/pipeline/parameters.py

```diff
@@ -0,0 +1,26 @@
+from dataclasses import dataclass
+from functools import cached_property
+
+from solidago.comparisons_to_scores import ContinuousBradleyTerry
+
+
+@dataclass
+class PipelineParameters:
+    # comparisons to scores
+    alpha: float = 0.02
+    r_max: float = 10.0
+
+    # collaborative scaling and aggregation
+    max_squashed_score: float = 100.0
+    score_shift_W: float = 1.
+    score_shift_quantile: float = 0.15
+    score_deviation_quantile: float = 0.9
+    W: float = 10.0
+
+    # voting rights
+    vote_weight_public_ratings: float = 1.0
+    vote_weight_private_ratings: float = 0.5
+
+    @cached_property
+    def indiv_algo(self):
+        return ContinuousBradleyTerry(r_max=self.r_max, alpha=self.alpha)
```

Comment:
In the ICML paper, I would like to further stress the fact that our pipeline is modular.
It would be nice here to also put as parameters other steps of the algorithms as ```@cached_property```, such as:
- Pretrusts & vouches to trust scores (which requires also exporting ```is_pretrusted``` and ```vouches.csv```)
- Trust scores to voting rights
- Contributor's comparison to individual scores
- Individual score rescaling
- Aggregation of individual scaled scores into global scores
- Scores to displayed scores

For readability, maybe the parameters should be directly defined in the corresponding methods.

Reply:
So you would suggest to create abstractions for the different steps of the algorithm, similar to the `ComparisonsToScoresAlgorithm` abstract class, which is implemented by both `HookeIndividualScores` and `ContinuousBradleyTerry` ?
https://github.com/tournesol-app/tournesol/blob/c6419a81667895aad8ad285f21cc313fcb291c00/solidago/src/solidago/comparisons_to_scores/base.py#L7-L35

That sounds good to me, but will require a bit more refactoring :)